sockeye/utils.py

```diff
@@ -700,12 +702,49 @@ def cleanup_params_files(output_folder: str, max_to_keep: int, checkpoint: int,
     """
     if max_to_keep <= 0:
         return
+
+    #
+    # make sure we keep N best params files from .metrics file according to strategy.
+    #
+
+    top_n = []
+    metrics_path = os.path.join(output_folder, C.METRICS_NAME)
+    
+    if max_params_files_to_cache > 0 and os.path.exists(metrics_path):
+        metric = cache_metric
+        strategy = cache_strategy
+            
+        maximize = C.METRIC_MAXIMIZE[metric]
+        points = get_validation_metric_points(model_path=output_folder, metric=metric)
+
+        if strategy == "best":
+            # N best scoring points
+            top_n = average._strategy_best(points, max_params_files_to_cache, maximize)
+
+        elif strategy == "last":
+            # N sequential points ending with overall best
+            top_n = average._strategy_last(points, max_params_files_to_cache, maximize)
+
+        elif strategy == "lifespan":
+            # Track lifespan of every "new best" point
+            # Points dominated by a previous better point have lifespan 0
+            top_n = average._strategy_lifespan(points, max_params_files_to_cache, maximize)
+        else:
+            raise RuntimeError("Unknown strategy, options: [best, last, lifespan]")
+
+    top_n = [x[1] for x in top_n]
```

Comment:
could this be made a set for faster membership checking?

Reply:
Sure. Will address.